src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs

```diff
@@ -0,0 +1,1116 @@
+            var initializer = operation.Initializer;
+
+            if (initializer != null)
+            {
+                children = children.Concat(new[] { initializer });
+            }
+
+            AssertEx.Equal(children, operation.Children);
+        }
+
+        public override void VisitVariableDeclaration(IVariableDeclarationOperation operation)
+        {
+            Assert.Equal(OperationKind.VariableDeclaration, operation.Kind);
+            IEnumerable<IOperation> children = operation.Declarators;
+            var initializer = operation.Initializer;
+
+            if (initializer != null)
+            {
+                children = children.Concat(new[] { initializer });
+            }
+
+            AssertEx.Equal(children, operation.Children);
+        }
+
+        public override void VisitSwitch(ISwitchOperation operation)
+        {
+            Assert.Equal(OperationKind.Switch, operation.Kind);
+            AssertEx.Equal(new [] { operation.Value}.Concat(operation.Cases), operation.Children);
+        }
+
+        public override void VisitSwitchCase(ISwitchCaseOperation operation)
+        {
+            Assert.Equal(OperationKind.SwitchCase, operation.Kind);
+            AssertEx.Equal(operation.Clauses.Concat(operation.Body), operation.Children);
+        }
+
+        public override void VisitSingleValueCaseClause(ISingleValueCaseClauseOperation operation)
+        {
+            Assert.Equal(OperationKind.CaseClause, operation.Kind);
+            Assert.Equal(CaseKind.SingleValue, operation.CaseKind);
+
+            // It is unexpected to have null operation.Value. The conditional logic is added
+            // as a work around for https://github.com/dotnet/roslyn/issues/23795 and should be removed
+            // once the issue is fixed
+            if (operation.Value == null)
+            {
+                Assert.Equal(LanguageNames.VisualBasic, operation.Language);
+                Assert.Empty(operation.Children);
+            }
+            else
+            {
+                AssertEx.Equal(new[] { operation.Value }, operation.Children);
+            }
+        }
+
+        public override void VisitRelationalCaseClause(IRelationalCaseClauseOperation operation)
+        {
+            Assert.Equal(OperationKind.CaseClause, operation.Kind);
+            Assert.Equal(CaseKind.Relational, operation.CaseKind);
+            var relation = operation.Relation;
+
+            AssertEx.Equal(new[] { operation.Value }, operation.Children);
+        }
+
+        public override void VisitDefaultCaseClause(IDefaultCaseClauseOperation operation)
+        {
+            Assert.Equal(OperationKind.CaseClause, operation.Kind);
+            Assert.Equal(CaseKind.Default, operation.CaseKind);
+            Assert.Empty(operation.Children);
+        }
+
+        private void VisitLocals(ILoopOperation operation)
+        {
+            VisitLocals(operation.Locals);
```

Comment:
This should actually never be hit, correct? ILoopOperation shouldn't ever be the base interface, so we should always hit a specific Visit.

Reply:
> This should actually never be hit, correct? ILoopOperation shouldn't ever be the base interface, so we should always hit a specific Visit.

This method is called explicitly below in VisitWhileLoop and the like.

---
In reply to: [157596257](https://github.com/dotnet/roslyn/pull/23840#discussion_r157596257) [](ancestors = 157596257)